packer/resources/features/mongo24/scripts/aws/helpers.rb

```diff
@@ -0,0 +1,82 @@
+require 'net/http'
+require 'json'
+require 'aws-sdk'
+
+module AwsHelper
+  class Non200Response < RuntimeError; end
+
+  FAILURES = [
+      Errno::EHOSTUNREACH,
+      Errno::ECONNREFUSED,
+      Errno::EHOSTDOWN,
+      Errno::ENETUNREACH,
+      SocketError,
+      Timeout::Error,
+      Non200Response,
+  ]
+
+  class InstanceData
+    def self.get_tags(instance_id = Metadata.instance_id)
+      ec2 = Aws::EC2::Client.new
+      tag_results = ec2.describe_tags({
+        filters: [{name: "resource-id", values: [instance_id]}]
+      }).tags
+      Hash[ tag_results.map{ |tag| [tag.key, tag.value]} ]
+    end
+  end
+
+  class Metadata
+    @@backoff = lambda { |num_failures| Kernel.sleep(1.2 ** num_failures) }
+    @retries = 3
+
+    def self.open_connection
+      http = Net::HTTP.new('169.254.169.254', 80, nil)
+      http.open_timeout = 5
+      http.read_timeout = 5
+      http.start
+      yield(http).tap { http.finish }
```

Comment:
I like this cheeky use of `tap`, although I don't think it will close the connection if the passed block raises an exception.